File: datasets/JPlag-Java/to-kotlin-intellij/Try.kt
package de.jplag.java

import java.io.File
import java.io.FileNotFoundException
import java.util.*

class Try {
    fun load(path: String?) {
        var scanner: Scanner? = null
        try {
            var other: Scanner // This is just here to keep the tokens similar.
            scanner = Scanner(File(path))
            while (scanner.hasNext()) {
                println(scanner.nextLine())
            }
        } catch (exception: FileNotFoundException) {
            exception.printStackTrace()
        } finally {
            scanner?.close()
        }
    }

    companion object {
        @JvmStatic
        fun main(args: Array<String>) {
            Try().load("DoesNotExist.txt")
        }
    }
}

File: datasets/JPlag-Java/to-typescript-convertai/Try.ts
import * as fs from 'fs';

class Try {
    public static main(args: string[]): void {
        new Try().load("DoesNotExist.txt");
    }

    public load(path: string): void {
        let scanner: fs.ReadStream | null = null;
        try {
            let other: fs.ReadStream; // This is just here to keep the tokens similar.
            scanner = fs.createReadStream(path);
            scanner.on('data', (chunk: Buffer) => {
                console.log(chunk.toString());
            });
            scanner.on('error', (err: Error) => {
                console.error(err);
            });
        } catch (exception) {
            console.error(exception);
        } finally {
            if (scanner !== null) {
                scanner.close();
            }
        }
    }
}

Try.main(process.argv.slice(2));

